pkg/agent/cache/accessrequest.go

```diff
@@ -44,11 +45,13 @@ func (c *cacheManager) GetAccessRequestByAppAndAPI(appName, remoteAPIID, remoteA
 	c.ApplyResourceReadLock()
 	defer c.ReleaseResourceReadLock()
 
-	accessRequest, _ := c.accessRequestMap.GetBySecondaryKey(appName + ":" + remoteAPIID + ":" + remoteAPIStage)
-	if accessRequest != nil {
-		ri, ok := accessRequest.(*mv1.AccessRequest)
-		if ok {
-			return ri
+	cachedAccessRequest, _ := c.accessRequestMap.GetBySecondaryKey(appName + ":" + remoteAPIID + ":" + remoteAPIStage)
+	if cachedAccessRequest != nil {
+		if ri, ok := cachedAccessRequest.(*v1.ResourceInstance); ok {
+			accessRequest := &mv1.AccessRequest{}
+			if accessRequest.FromInstance(ri) == nil {
```

Comment:
If it's nil you wanna return? Or != nil? Slightly confused. No code in front of me.

 Still 🏃🏃🏃

Reply:
FromInstance returns an error only. The object that calls it is loaded to from the resource instance that was sent in.

Reply:
👍🏾